### PR TITLE
April 1 patch

### PR DESCRIPTION
**Pull request by edjubuh**
- Creating projects now properly respect template ignore files
- Packet retry patch does a deep copy as intended
- Upgrade command for Windows produces a reducedgui for better feedback when running it

### CODE REVIEW

**Comment by edjubuh** [comment]
This will release CLI 2.5.1. Test plan as follows:

- Upload over VEXnet 2.0 using `-vd` flags and check to see if packets are successfully retried with the message `RETRYING PACKET AT WRITE`
- Create a project. 
    - Verify all files are correct
    - Make and upload it.
- Upgrade a project
     - Verify all files that intended to get modified get modified

**Review by jwalk511 — APPROVED** [review]
`conduct upgrade` successfully upgrades `common.mk`, `libpros.a`, and `api.h`.
`conduct new` successfully creates a new, makable project.
`flash -vd` wrote everything just fine with 0 packet loss from 20 feet away over VEXnet 2.0, and thus could not be checked. But the flasher actually works, so I approve anyway.

**Comment by HotelCalifornia** [comment]
`pros flash` over VEXnet 2.0 across the room resulted in a lot of failed packets, all of which were retried and (eventually) sent successfully. I can't say much for its speed, but it seems to work pretty well.

[here's a copy of the output if you're interested](https://gist.github.com/HotelCalifornia/6b3d1e35c9574aaf8cb8d76933cca41d)

`pros conduct new` successfully created a project ~~(`pros conduct info-project` shows it's on kernel 2.11.0, and it thinks the kernel is up-to-date-- not sure if that's intentional at this stage)~~, which I successfully compiled and flashed (once VEXnet decided it would connect).

~~I don't have any old projects with which to test `pros conduct upgrade`, and as I mentioned above, I couldn't get the CLI conductor to recognize that there were any templates other than 2.11.0 on `pros-mainline`.~~

`pros conduct ls-template` correctly showed that there was a new kernel available for download, and `pros conduct download latest kernel` downloaded it. From there, `pros conduct upgrade` successfully changed `common.mk`, `libpros.a`, and `API.h` as verified by running `diff`.

```
$ diff test0/include/API.h test1/include/API.h
1580a1581,1588
> /**
>  * Enables IWDG watchdog timer which will reset the cortex if it locks up due to static shock
>  * or a misbehaving task preventing the timer to be reset. Not recovering from static shock
>  * will cause the robot to continue moving its motors indefinitely until turned off manually.
>  *
>  * This function should only be called once in initializeIO()
>  */
> void watchdogInit();
```

**Review by HotelCalifornia — APPROVED** [review]

File: prosflasher/bootloader.py [hunk 1]
```diff
@@ -155,7 +159,7 @@ def write_flash(port, start_address, data, retry=2):
     debug('STM BL RESPONSE TO WRITE: {}'.format(response))
     if response is None or len(response) < 1 or response[0] != ACK:
         if retry > 0:
-            debug('RETRYING PACKET')
+            debug('RETRYING PACKET AT WRITE')
             write_flash(port, start_address, data, retry=retry - 1)
```
Comment (edjubuh) [inline prosflasher/bootloader.py]: should have a return here (and in the other places)

**Comment by HotelCalifornia** [comment]
Confirmed flasher is working after 6dd0c65

**Pull request merged**